tests/util.rs

```diff
@@ -66,3 +75,73 @@ impl TsGenerator {
         self.ts_pool.next().unwrap()
     }
 }
+
+/// A logger that add a test case tag before each line of log.
+struct CaseTraceLogger {
+    level: LogLevelFilter,
+    f: Option<Mutex<File>>,
+}
+
+impl Log for CaseTraceLogger {
+    fn enabled(&self, meta: &LogMetadata) -> bool {
+        meta.level() <= self.level
+    }
+
+    fn log(&self, record: &LogRecord) {
+        if self.enabled(record.metadata()) {
+            let tag = util::get_tag_from_thread_name().unwrap_or("".into());
+            let t = time::now();
+            // TODO allow formatter to be configurable.
+            let _ = if let Some(ref out) = self.f {
+                write!(out.lock().unwrap(),
+                       "{} {},{:03} {}:{} - {:5} - {}\n",
+                       tag,
+                       time::strftime("%Y-%m-%d %H:%M:%S", &t).unwrap(),
+                       t.tm_nsec / 1000_000,
+                       record.location().file().rsplit('/').nth(0).unwrap(),
+                       record.location().line(),
+                       record.level(),
+                       record.args())
+            } else {
+                write!(io::stderr(),
+                       "{} {},{:03} {}:{} - {:5} - {}\n",
+                       tag,
+                       time::strftime("%Y-%m-%d %H:%M:%S", &t).unwrap(),
+                       t.tm_nsec / 1000_000,
+                       record.location().file().rsplit('/').nth(0).unwrap(),
+                       record.location().line(),
+                       record.level(),
+                       record.args())
+            };
+        }
+    }
+}
+
+impl Drop for CaseTraceLogger {
+    fn drop(&mut self) {
+        if let Some(ref w) = self.f {
+            w.lock().unwrap().flush().unwrap();
+        }
+    }
+}
+
+// A help function to initial logger.
+pub fn init_log() {
+    let output = env::var("LOG_FILE").ok();
+    if output.is_none() {
+        return;
```

Comment:
so if I want to enable log in my own compute and output to console, it doesn't work too, only if I use a log file and set this env?